### DISCUSSION TITLE
Using nested profiles with the new suites?

### DISCUSSION THREAD

**Discussion by lycheese** [nrdxp/nixdev#346] [Q&A] [answered]
Up until now I have used profiles in the following manner:

```
dev = [ core develop graphical network users.lycheese ];
gaming = [ graphical.gaming ];
hidpi = [ graphical.hidpi ];
```

Where graphical.hidpi points to a subdirectory of graphical that is not imported by `graphical/default.nix` (same for gaming). After updating devos and switching to the new `importables` way of defining suites, I got the following error:

```
error: value is a path while a set was expected

       at /nix/store/ksgnlpfflgp2gqq4i9xnp2v2hxs5l7lb-source/flake.nix:98:24:

           97|                     graphical network users.lycheese ];
           98|             gaming = [ graphical.gaming ];
             |                        ^
           99|             hidpi = [ graphical.hidpi ];
(use '--show-trace' to show detailed location information)
```

I put hidpi and gaming into the `profiles` folder for now but if possible I would like to return them into `profiles/graphical` while being able to import them into suites.

Edit-3:
When combining suites in a hosts declaration with `imports = suites.dev ++ suites.gaming;` I am getting a stack overflow. Is there a special syntax for combining the new suites?

<details><summary>Unrelated</summary>
Edit:
Also now getting this error when running `flk <host> switch`:

```
error: value is null while a set was expected

       at /nix/store/v9hyb5cjk6y3w97rp320h3xp246p26nr-source/flake.nix:52:19:

           51|                 moduleDeclarationFile =
           52|                   (builtins.unsafeGetAttrPos "modules" args).file;
             |                   ^
           53|
(use '--show-trace' to show detailed location information)
error: getting status of '/tmp/nixos-rebuild.Cur29H/result': No such file or directory
```

Edit-2:
Known problem: https://github.com/divnix/digga/issues/87

</details>

**Comment by blaggacao** [comment, marked as answer]
The new `rakeLeaves` implementation achieves being more universal than the previous implementation, for example , with slight modifications, `rakeLeaves` is being used for `hosts` and `modules`, as well. Thus it promotes clarity and reusability. One complaint we had in the past is that the mental model for folder imports has varied too much.

Now, `rakeLeaves` recursively walkes through folders and rakes leaves. Leaves are: 1) files ending in `nix` or 2) folders that contain a `default.nix` file.

Those leaves are raked into a nested attribute set according to their place in the folder hierarchy.

The general solution to keep the organization primitives that previously where implicitly allowed is to make an implicit common profile explicit by placing it into (in your case) `graphical/common.nix`.

I can't say much about the stack overflow error. At first sight, it seems foreign to `devos` / `digga`. But who knows...
